File: 2025/day-02/src/bin/part1.rs
fn main() {
    let mut lines: Vec<&str> = include_str!("../inputs/input.txt").lines().collect();
    let mut safe_reports = 0;
    for line in lines {
        let nums: Vec<_> = line
            .split_whitespace()
            .map(|c| c.parse::<i32>().unwrap())
            .collect();
        println!("{:?}", nums);
        let mut windows = nums.windows(2).peekable();
        let mut safe_report = true;
        let mut decreasing: bool = true;
        if let Some(pair) = windows.peek() {
            if pair[0] < pair[1] {
                decreasing = false;
            } else {
                decreasing = true;
            }
        }

        while let Some(pair) = windows.next() {
            let this_window_decreasing: bool;
            if pair[0] < pair[1] {
                this_window_decreasing = false;
            } else {
                this_window_decreasing = true;
            }
            if decreasing != this_window_decreasing || pair[0] == pair[1] {
                safe_report = false;
                break;
            }
            if pair[0].abs_diff(pair[1]) > 3 {
                safe_report = false;
                break;
            }
        }
        if safe_report {
            println!("{:?}", line);
            safe_reports += 1;
        }
    }
    println!("{:?}", safe_reports);
}

File: 2016/day-02/part1.py
with open('input.txt', 'r') as file:
    lines = [line.strip() for line in file]
    
x = 1
y = 1
map = {
    (0, 0): 1,
    (1, 0): 2,
    (2, 0): 3,
    (0, 1): 4,
    (1, 1): 5,
    (2, 1): 6,
    (0, 2): 7,
    (1, 2): 8,
    (2, 2): 9,
}

code = ""
for line in lines:
    for direction in line:
        if direction == "U":
            y -= 1
        elif direction == "L":
            x -= 1
        elif direction == "R":
            x += 1
        elif direction == "D":
            y += 1
        if x < 0:
            x = 0
        if y < 0:
            y = 0
        if x > 2:
            x = 2
        if y > 2:
            y = 2
    code += str(map[(x, y)])
    
print(code)